header layout initial draft / DPavliuk

DmytroP888 opened the pull request:
header development

DmytroP888 committed:
header layout initial draft / DPavliuk

DmytroP888 committed:
Name correction

nastiaymlnv commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,74 @@
+margin-left: 16px;
+font-family: 'Montserrat-700';
+font-style: normal;
+font-weight: 700;
```
Rows 25 and 26 are useless, because font-family provides its style and weight by default

DmytroP888 replied:
Of course I'll fix it.

nastiaymlnv commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,74 @@
+justify-content: space-between;
+`;
+
+export const NavheaderBlock = styled.div` 
```
Try using more semantic tags instead of div

DmytroP888 replied:
I will try to do this.

nastiaymlnv commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,74 @@
+width: 90px;
+min-height: 15px;
+margin-top: 1px;
+font-family: "Montserrat Medium";
```
"Montserrat Medium" is a little bit incorrect name. Name it "Montserrat-Regular" (you can look into fonts.jsx for fonts names you need :) )

DmytroP888 replied:
Ok.

nastiaymlnv commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,74 @@
+min-height: 15px;
+margin-left: 15px;
+margin-top: 1px;
+font-family: "Montserrat Medium";
```
Right here with font name and font-weight too

DmytroP888 replied:
Thanks for the advice)) I fixed everything)

nastiaymlnv commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,74 @@
+justify-content: space-between;
+align-items: center;
+background-color: white;
+box-shadow: 0px 2px 4px #00000026;
```
I think the color code has redundant digits (it may be 6)

DmytroP888 replied:
I will check

DmytroP888 committed:
Merge branch 'master' of https://github.com/vladhybalo/kids-awards into header-layout-initial-draft-/-DPavliuk

DmytroP888 committed:
Complete change of header structure.

1. Installed plugin - vite-plugin-svgr - https://www.npmjs.com/package/vite-plugin-svgr
2. Changed the vite.config file

3. Installed library - npm and cyrillic-to-translit-js - https://www.npmjs.com/package/cyrillic-to-translit-js

4. The file - src\components\Header\Header.jsx contains two temporary hardcoded constants UserBalance - the balance of points, UserName - the name of the user.

nastiaymlnv committed:
add Spinner component and hoc, paste them into pages (#14)

* add Spinner component and hoc, paste them into pages

* add constant color

nastiaymlnv committed:
Auth page (#12)

* change login path name in Router.jsx, rename Login.jsx into AuthPage.jsx

* start AuthPage, add AuthForm without logic

* add adaptive AuthPage with AuthForm without logic

* fix footer adaptivity

* add input validation in AuthForm, returning obj to parent component

* add binding to store

* fix deploy error

* fix error with input credentials validation, add more error messages

* fix all remarks

* modify input validation with onChange

* remove switch from different async actions

* minor fixes with naming, images etc, add alert for failed registration

* translate UI into English

* add redirect to home after successful sign in/up

nastiaymlnv committed:
Auth page (#12)

* change login path name in Router.jsx, rename Login.jsx into AuthPage.jsx

* start AuthPage, add AuthForm without logic

* add adaptive AuthPage with AuthForm without logic

* fix footer adaptivity

* add input validation in AuthForm, returning obj to parent component

* add binding to store

* fix deploy error

* fix error with input credentials validation, add more error messages

* fix all remarks

* modify input validation with onChange

* remove switch from different async actions

* minor fixes with naming, images etc, add alert for failed registration

* translate UI into English

* add redirect to home after successful sign in/up

andriiivaniuk committed:
add BaseModal component (#13)

* add BaseModal component

* fixed according to comments

* quick fix

* final BaseModal changes according to comments

andriiivaniuk committed:
add BaseModal component (#13)

* add BaseModal component

* fixed according to comments

* quick fix

* final BaseModal changes according to comments

andriiivaniuk commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+)
+const PointBalUser = () => (
+    <PointsBalance>
+        Баланс баллов:
+        <NumberBalance>{UserBalance}</NumberBalance>
```
We have agreed to get rid of russian language from our project :)
Could you please translate all the titles to English?

DmytroP888 replied:
Thank you) I'm starting to fix

andriiivaniuk commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,289 @@
+    }
+`;
+
+// -------------------------------------------------
+export const MobileHamburgerbackground = styled.nav`
```
commented out code should be deleted

DmytroP888 replied:
Yes, exactly, I will remove it from the code)

andriiivaniuk commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,289 @@
+        }
+    }
+`;
+
+
+
+
+
+
```
too many empty lines in a row

DmytroP888 replied:
Oh sure)

andriiivaniuk commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,289 @@
+    right: 20px;
+    padding-left: 11px;
+    position: absolute;
+    &:hover {
+        box-shadow: 
+        inset rgb(0 0 0 / 19%) 0px 0px 10px 0px,
+        rgb(0 0 0 / 19%) 1px 3px 10px 0px;
+        border-radius: 50px;
+        padding-left: 10.5px;
+        padding-top: 1.2px;
+        cursor: pointer;
```
this custom cross animation is cool, but I think we should strictly adhere to our mockup

DmytroP888 replied:
I will remove this hover)

andriiivaniuk commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+                                        <MobileHamburgerCross1 />
+                                    </MobileHamburgerCross>
+                                </MobileHamburgerCrossBox>
+                                <NavLink to="home" onClick={toggle}
+                                    style={({ isActive }) => { return { color: isActive ? 'black' : 'white' } }}>
+                                    <MobileHamburgerSectionMenuUp>Главная</MobileHamburgerSectionMenuUp>
+                                </NavLink>
+                                <NavLink to="planning" onClick={toggle}
+                                    style={({ isActive }) => { return { color: isActive ? 'black' : 'white' } }}>
+                                    <MobileHamburgerSectionMenu>Планирование</MobileHamburgerSectionMenu>
+                                </NavLink>
+                                <NavLink to="awards" onClick={toggle}
+                                    style={({ isActive }) => { return { color: isActive ? 'black' : 'white' } }}>
+                                    <MobileHamburgerSectionMenu>Награды</MobileHamburgerSectionMenu>
+                                </NavLink>
+                                <NavLink to="contacts" onClick={toggle}
+                                    style={({ isActive }) => { return { color: isActive ? 'black' : 'white' } }}>
+                                    <MobileHamburgerSectionMenu>Контакты</MobileHamburgerSectionMenu>
+                                </NavLink>
```
this code looks too repetitive, I think React gives us a possibility to elegantly lower code lines amount using Array.map method

DmytroP888 replied:
Thanks for the advice) I'll try to change the approach using the map.

andriiivaniuk commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,289 @@
+    align-items: center;
+    padding-left: 20px;
+    ${fontRegular}
+    &:hover {
+    box-shadow: 
+    rgb(0 0 0 / 19%) 1px 3px 10px 0px, 
+    inset -6px -6px 8px rgb(255 255 255 / 80%), 
+    inset 6px 6px 8px rgb(70 70 70 / 12%);
+  }
+`;
```
please keep hover styles according to mockup

DmytroP888 replied:
I will remove this hover.

andriiivaniuk commented:
It looks like according to mockup after activating hamburger menu on tablet view, main area should become faded, please see pictures
![Capture1](https://user-images.githubusercontent.com/96084610/184707326-da74d878-1fd9-49bb-918e-4698299877ed.JPG)
![Capture2](https://user-images.githubusercontent.com/96084610/184707328-e32a8f46-aa4a-48f4-af5b-53e14a2c11e0.JPG)

nastiaymlnv commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+import CyrillicToTranslit from 'cyrillic-to-translit-js'
+import ListenerWidth from "./ListenerWidth"
+import {
+    ContainerHeder, LogoHeader, LeftBlock, PointsBalance, NumberBalance,
```
If the content doesn't fit the screen, it's more readable to make each word from a new row

DmytroP888 replied:
Thanks for the lesson) I'll fix it. I will try to write like this in the future.

nastiaymlnv commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,289 @@
+    font-family: 'Montserrat-700';
+    font-size: 16px;
+    letter-spacing: 0.08em;
+    color: #000000;
```
It will be good if you use color constants because most colors are reused in other places. You can get some constants and create a new one in assets/themes/colors.js (this file you can find in the project after merging the AuthForm branch into the master branch)

DmytroP888 replied:
Thanks for the advice Nastia) I'll try to do it.

DmytroP888 commented:
> It looks like according to mockup after activating hamburger menu on tablet view, main area should become faded, please see pictures ![Capture1](https://user-images.githubusercontent.com/96084610/184707326-da74d878-1fd9-49bb-918e-4698299877ed.JPG) ![Capture2](https://user-images.githubusercontent.com/96084610/184707328-e32a8f46-aa4a-48f4-af5b-53e14a2c11e0.JPG)

Really. I got very carried away by UI UX and missed the obvious things. That's my fault. I will correct it.

vladhybalo commented on src/App.jsx:
```diff
@@ -4,12 +4,15 @@ import GlobalFonts from './assets/fonts/fonts';
       <GlobalFonts />
       <GlobalStylesReset />
-      
+
```
Redundant empty string

DmytroP888 replied:
Ok)

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+import React, { useState } from "react"
+import { useLocation } from 'react-router-dom'
+import { NavLink } from "react-router-dom"
```
You could use one line to import all what you want from react router

DmytroP888 replied:
Oops, really. I'll fix it)

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+import { ReactComponent as ExitIconMobile } from "../../assets/image/HeaderNavExitIconMobile.svg"
+
+const UserBalance = 123
+const UserName = 'UserName'
```
Now you can get this data from store as auth page is into master

DmytroP888 replied:
I'll try to figure it out and connect the data correctly)

vladhybalo replied:
Its a correct way for prototype, but now we can do full integration)

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+const UserName = 'UserName'
+const logOut = () => localStorage.clear()
+
+const cyrillicToTranslit = new CyrillicToTranslit()
```
What the hell is that?

DmytroP888 replied:
Well, it's sort of... a way to log out of an account. I just didn't understand at first how else it could be done. 🙄

vladhybalo replied:
I can't get how transit can help you to log out? Anycase you should use data received from the server and pls don't care about its language)

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+const Header = () => {
+    const [isOpen, setIsOpen] = useState(false)
+    const toggle = () => setIsOpen(!isOpen)
+    const width = (ListenerWidth() - 1)
```
PaskalCase names suits for components, not for functions.

DmytroP888 replied:
Thanks for the tips. I'll replace it with  -  useScreenWidth      )))

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+
+const cyrillicToTranslit = new CyrillicToTranslit()
+const firstLetter = UserName && (UserName.charAt(0))
+const nameLetter = cyrillicToTranslit.transform(firstLetter).toUpperCase()
```
It is easier to convert by CSS property text-transform

DmytroP888 replied:
Thanks for the advice) I will study the possibility of such an implementation.

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+        </LogoHeader>
+    </NavLink>
+)
+const MenuAuthDeskTab = () => (
```
If you decided to make this code a separate component it could be placed into separete file

DmytroP888 replied:
Can I create a file called Helper and add components like this to that file?)

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+        </>
+    )
+}
+export default Header
```
We received a huge component, I saw your previous implementation where some separete components like for loggedIn user or not and etc. Is it possible to split the code?

DmytroP888 replied:
Yes, of course, we can separate the components. And combine Guest and User in a common Header component. But, as it seems to me, in such cases there will be a lot of the same type of code in the Guest and User components ( ListenerWidth, common CSS Header blocks ...)

vladhybalo replied:
All common blocks could be also separate components) and CSS will be common. You can have into folder a "main" component - composition of some smaller components, smaller components themselves, and one big file with CSS.
Of course, you can also separate CSS files, its up to you) I just meant that huge CSS isnt such a problem like JS or JSX because its pretty simple: instruction and value, instruction and value, and so on)

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+                        <ContainerHeder>
+                            <LeftBlock>
+                                <LogoHead />
+                                {location.pathname === '/home' &&
```
pathname for home is `/`

DmytroP888 replied:
Ok)

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+                            </LeftBlock>
+                            <MenuHeaderBlock>
+                                <NavLink to="home"
+                                    style={({ isActive }) => { return { color: isActive ? 'black' : 'rgb(133, 133, 152)' } }}>
```
is it possible to avoid inline styles?

DmytroP888 replied:
Yes, of course) But this will lead to an increase in code. The Header.styled file will increase, and the number of imports will increase...

vladhybalo replied:
Yep this is a problem too, but we already have an overloaded component. It could be fixed by separating code on a few components

vladhybalo commented:
Why not logged in user see data for signedIn?
![image](https://user-images.githubusercontent.com/52242019/184953948-bc698fa0-9f67-47de-9a9b-61ef629b9305.png)

vladhybalo commented on src/components/Header/ListenerWidth.jsx:
```diff
@@ -0,0 +1,17 @@
+    const width = window.innerWidth
+    return width
+}
+export default function ListenerWidth() {
```
you have created your own hook! Cool! But general note, there is a convention to name hooks with use prefix. Like useScreenWidth

DmytroP888 replied:
A very very good note. A very good lesson for me. Thanks, I'll fix it )))

vladhybalo commented on src/components/Header/Header.styled.jsx:
```diff
@@ -0,0 +1,289 @@
+    Z-index: 11;
+    position: absolute;
+`;
+export const MobileHamburgerCross1 = styled.div`
```
`Cross` and `Cross1` arent well descriptive have.
If `Cross1` is cross himself and `Cross` is its wrapper you can name:
`Cross` -> `CrossHolder` or `CrossWrapper` 
`Cross1` -> `Cross`

DmytroP888 replied:
Yes of course thank you. I'll fix it))

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,255 @@
+                                        <MenuInitiaLetter>{nameLetter}</MenuInitiaLetter>
+                                    </MenuInitiaLetterCircle>
+                                    <MenuName>{UserName}</MenuName>
+                                    <NavLink to="login" onClick={`${toggle}${logOut}`}>
```
Its some hack to add 2 event handlers?

DmytroP888 replied:
I came up with this idea, I tried it, and it works. I don't know if it's the right thing to do. If it is permissible, then let it be a feature of our small team. If not, I'll fix it somehow and write differently))

DmytroP888 commented:
> Why not logged in user see data for signedIn? ![image](https://user-images.githubusercontent.com/52242019/184953948-bc698fa0-9f67-47de-9a9b-61ef629b9305.png)

At this point, the logic and the result of the server and Redax work according to the User's data have not yet been connected. At the moment, you just need to set the UserName constant to null in the Header file.

DmytroP888 committed:
some fixes

nastiaymlnv committed:
Auth page finalization (#15)

* add useSelector + useEffect instead of subscribe

* add signing in with google (get request error)

* add spinner while request is in progress

* fix remarks from review

* minor fixes

* remove console.log

nastiaymlnv committed:
Task List implementation  (#17)

* add card and different buttons (toggler, status, add btn)

* add propTypes, make separate component for defining card btn, and create enum for it

* remove redundant code from Awards.jsx

* rename AddCardBtn component into ChooseDayBtn

* fix naming, props passing & propTypes

DmytroP888 committed:
Corrections according to the review

DmytroP888 committed:
Merge branch 'master' of https://github.com/vladhybalo/kids-awards

DmytroP888 committed:
fixing commits

DmytroP888 committed:
Merge branch 'master' of https://github.com/vladhybalo/kids-awards into header-layout-initial-draft-/-DPavliuk

DmytroP888 committed:
clear

DmytroP888 committed:
Merge branch 'master' into header-layout-initial-draft-/-DPavliuk

DmytroP888 committed:
Updating the master and fixing commits

DmytroP888 committed:
For easy verification minor corrections have been

DmytroP888 commented:
For easy readability and verification - minor corrections have been made in order to highlight the files that were developed in this PullRequest.

Also, in order to be able to open and use the PR in firebase, some lines of code from the wizard were temporarily commented out, which are still in development and caused a loading error.

VladyslavHybalo commented on src/components/AddTask/AddTask.jsx:
```diff
@@ -1,17 +1,17 @@
 
 AddTask.propTypes = {
-    addDayToTask: PropTypes.func.isRequired
+    // addDayToTask: PropTypes.func.isRequired
```
uncommit it pls

DmytroP888 replied:
Yes, of course) I will remove the comment in the next Pool Request

vladhybalo commented on src/components/CardBtn/CardBtn.jsx:
```diff
@@ -1,13 +1,13 @@
 import React from "react";
-import PropTypes from 'prop-types';
+// import PropTypes from 'prop-types';
```
Pls use proptypes validation

DmytroP888 replied:
Of course, I will remove the comments from the code))

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,21 @@
+
+const Header = () => {
+
+    const stateUsInfouserData = useSelector(state => state.userInfo.userData)
```
IMHO this var isnt readable at all - `stateUsInfouserData`. I would name it userData

DmytroP888 replied:
Of course - I will replace it with userData) Thanks for the advice)

vladhybalo commented on src/components/Header/Header.jsx:
```diff
@@ -0,0 +1,21 @@
+    return (
+        <>
+            {!stateUsInfouserData && // ----- AUTH -------
+                <HeaderAuth />
```
LoggedOutHeader or UnAuthorizedHeader
 and in save way below

vladhybalo replied:
Could be done by ternary

DmytroP888 replied:
Ok)) I'll change it to a ternary expression.

vladhybalo commented on src/components/Header/HeaderAuth.jsx:
```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react"
+import { NavLink, useLocation } from 'react-router-dom'
+import useScreenWidth from "./ListenerWidth"
```
Pls add empty line

DmytroP888 replied:
Thanks for the answer. Now I have learned the rules for writing imports

vladhybalo commented on src/components/Header/HeaderAuth.jsx:
```diff
@@ -0,0 +1,92 @@
+    MobileHamburgerSectionMenuUp,
+    MobileHamburgerSectionMenu
+} from "./Header.styled"
+import { MAIN_DARK, MAIN_BACKGROUND } from "../../assets/themes/colors"
```
themes imports should be between libraries and components

DmytroP888 replied:
Oh sorry, I'll change the order of the imports)

vladhybalo commented on src/components/Header/HeaderAuth.jsx:
```diff
@@ -0,0 +1,92 @@
+
+    return (
+        <>
+            {width >= 769 &&                    // --------------- Desktop -------------
```
I will be more semantic if it done by media queries

DmytroP888 replied:
Yes)) I will remove this hook and use CSS

vladhybalo commented on src/components/Header/HeaderUser.jsx:
```diff
@@ -0,0 +1,201 @@
+import React, { useState } from "react";
+import { useSelector, useDispatch } from "react-redux"
+import { NavLink, useLocation } from 'react-router-dom'
+import CyrillicToTranslit from 'cyrillic-to-translit-js'
```
pls remove this library

DmytroP888 replied:
Oh yes, indeed, I completely missed the point that we now use the first letter of email. And there is no Cyrillic alphabet in the mail name))))) THANK YOU

vladhybalo commented on src/components/Header/HeaderUser.jsx:
```diff
@@ -0,0 +1,201 @@
+                        <MobileHamburgerDarkened >
+                        </MobileHamburgerDarkened>
+                    }
+                    <MobileHamburgerbackground Active={isOpen}>
```
To make this component smaller you could make some standalone components inside this folder.
Like: Hamburger, Navigation, PointsBalance and etc

DmytroP888 replied:
Most likely I can separate the following blocks into separate components:
- MobileHamburgerLoginBox
- MobileHamburgerCrossBox
- Navigation (NavLink)

vladhybalo commented on src/components/Header/ListenerWidth.jsx:
```diff
@@ -0,0 +1,18 @@
+        return () => window.removeEventListener('resize', handleResize);
+    }, []);
+    return windowDimensions;
+}
```
If you will succeed with css pls remove this hook)

DmytroP888 replied:
Yes)) I will remove this hook and use CSS

vladhybalo commented on src/ducks/userInfo/userInfoActions.jsx:
```diff
@@ -79,4 +87,17 @@ export const fetchGoogleData = () => {
     }
+}
+
+export const fetchLogOut = () => {
```
I think logout is one word. fetchLogout

DmytroP888 replied:
Oh sure. 😳
Thanks))

vladhybalo commented on vite.config.js:
```diff
@@ -1,7 +1,9 @@
 import { defineConfig } from 'vite'
 import react from '@vitejs/plugin-react'
+import svgr from "vite-plugin-svgr"
```
We decided to remove it, didnt we?

DmytroP888 replied:
Yes, of course, I just didn't know that this line should be removed)) I no longer use this library in components. I import directly SVG. Thanks for checking and advice :)

DmytroP888 committed:
Merge branch 'master' of https://github.com/vladhybalo/kids-awards into header-layout-initial-draft-/-DPavliuk

DmytroP888 committed:
new

DmytroP888 committed:
review fixes

DmytroP888 committed:
Removing an extra file 

Removing an extra file (draft, accidental commit from another branch)

vladhybalo commented on src/components/Header/MobileMenu.jsx:
```diff
@@ -0,0 +1,68 @@
+import ExitIconMobile from "../../assets/image/Header-exit-icon-mobile.svg"
+
+const MobileMenu = () => {
+    let userName = 'email'
```
const userName = userData ? userData.email ? null;
const nameLetter = userName ? userName.charAt(0) : null

DmytroP888 replied:
Thanks for the professional advice)) Everything is fixed

![image](https://user-images.githubusercontent.com/96341034/189455139-e71a70fc-3f99-47fe-af5d-2312faed6a16.png)

vladhybalo commented on src/components/Header/MobileMenu.jsx:
```diff
@@ -0,0 +1,68 @@
+    }
+    const nameLetter = userName && (userName.charAt(0))
+    const [isOpen, setIsOpen] = useState(false)
+    const toggle = () => { setIsOpen(!isOpen) }
+    const childToggle = () => { setIsOpen(!isOpen) }
```
As there is no difference between them you can use one function

DmytroP888 replied:
Right now I am using one function. Indeed, they were identical. Thanks for checking))
![image](https://user-images.githubusercontent.com/96341034/189454795-ba51832e-0de6-40a6-a443-9f6cdf534d20.png)

vladhybalo commented on src/components/Header/MobileMenu.jsx:
```diff
@@ -0,0 +1,68 @@
+                        </NavLink>
+                    </MenuHeaderBlock>
+                </MobileHamburgerLoginBox>
+                <Navigation func={childToggle} />
```
func is a bad name, it should be named same as property you pass

DmytroP888 replied:
function renamed to - hideMobileмenu

vladhybalo commented on src/components/Header/MobileMenu.jsx:
```diff
@@ -0,0 +1,68 @@
+    const nameLetter = userName && (userName.charAt(0))
+    const [isOpen, setIsOpen] = useState(false)
+    const toggle = () => { setIsOpen(!isOpen) }
+    const childToggle = () => { setIsOpen(!isOpen) }
```
isOpen isnt specific enough, maybe name it menuOpened?

DmytroP888 replied:
State name changed - menuOpened )

vladhybalo commented on src/components/Header/MobileMenu.jsx:
```diff
@@ -0,0 +1,68 @@
+    }
+    const nameLetter = userName && (userName.charAt(0))
+    const [isOpen, setIsOpen] = useState(false)
+    const toggle = () => { setIsOpen(!isOpen) }
```
toggle -> toggleMenu?

DmytroP888 replied:
Changed the name of the function. Thanks for the advice))
![image](https://user-images.githubusercontent.com/96341034/189454269-988c49a7-b94f-4bc9-8a20-c057ea25e6fd.png)

vladhybalo commented on src/components/Header/NameExiticon.jsx:
```diff
@@ -0,0 +1,38 @@
+    let userName = 'email'
+    const dispatch = useDispatch()
+    const logout = () => dispatch(fetchLogout(formData))
+    const userData = useSelector(state => state.userInfo.userData)
+    if (userData) {
+        userName = userData.email
+    }
+    const nameLetter = userName && (userName.charAt(0))
```
Copy&paste from component above, maybe you need to pass this data from some parent component or separate them in another way

DmytroP888 replied:
All redone as you said. I removed these
(repeating) blocks from two files (MobileMenu, Logouticon). And instead included this block in the HeaderUser parent. From where, with the help of props, I passed all the necessary properties ( props.userName, props.nameLetter, and the logoutProps function).
Thank you for the lesson))

![image](https://user-images.githubusercontent.com/96341034/189454058-5bef23d7-deb1-430f-bc8d-131075e0409b.png)

vladhybalo commented on src/components/Header/NameExiticon.jsx:
```diff
@@ -0,0 +1,38 @@
+} from "./Header.styled"
+import ExitIcon from "../../assets/image/Header-exit-icon.svg"
+
+const NameExiticon = () => {
```
I can understand from the name what does this component do

DmytroP888 replied:
Component renamed to - Logauticon

DmytroP888 replied:
Component renamed to - Logouticon

VladyslavHybalo replied:
LogoutIcon)

vladhybalo commented on src/components/Header/PointsBal.jsx:
```diff
@@ -0,0 +1,20 @@
+    let userBalance = 0
+    const userData = useSelector(state => state.userInfo.userData)
+    if (userData) {
+        userBalance = userData.balance
```
pls get balance from store directly

DmytroP888 replied:
changed like this:

let userBalance = 0
    const userAuthorized = store.getState().userInfo.token
    if (userAuthorized) {
        userBalance = store.getState().userInfo.userData.balance
    }

VladyslavHybalo replied:
Could you do it without let?

DmytroP888 replied:
Unfortunately, in this case I get an error. What should I do?

DmytroP888 replied:
I tried to replace let with const, but then it looks like something is changing with the scope

DmytroP888 replied:
Yes, I fixed everything. Thank you for the lesson))
![image](https://user-images.githubusercontent.com/96341034/189453378-fd4e166b-0db5-438f-9353-6b37a7e0ed9a.png)

vladhybalo commented on src/ducks/userInfo/userInfoActions.jsx:
```diff
@@ -1,9 +1,10 @@
-import axios from 'axios';
+import axios from 'axios'
```
prev version with semicolon I liked more

DmytroP888 replied:
Yes, of course, done))

DmytroP888 committed:
Fixed previous commit errors

DmytroP888 committed:
Merge branch 'header-layout-initial-draft-/-DPavliuk' of https://github.com/vladhybalo/kids-awards into header-layout-initial-draft-/-DPavliuk

DmytroP888 committed:
Remarks processing

vladhybalo commented on src/components/Header/Logouticon.jsx:
```diff
@@ -0,0 +1,35 @@
+import ExitIcon from "../../assets/image/Header-exit-icon.svg"
+
+const Logouticon = (props) => {
+    const logoutAction = () => { props.logoutProps() }
```
I dont see a reason for this function, you can use logout directly
PS and dont call it logoutProps, just `logout`, or if you need to have a logout function and logout action creator you can call it  `logout `and `logoutAction`

DmytroP888 replied:
I removed the props passing from this file and the logout click event from the parent. Now this function and event are in this file))
![Picture1](https://user-images.githubusercontent.com/96341034/189674044-e972779b-7222-430d-bfa7-339144cada85.png)

vladhybalo commented on src/components/Header/MobileMenu.jsx:
```diff
@@ -0,0 +1,67 @@
+import ExitIconMobile from "../../assets/image/Header-exit-icon-mobile.svg"
+
+const MobileMenu = (props) => {
+    const logoutAction = () => { props.logoutProps() }
```
same here

DmytroP888 replied:
I removed the props passing from this file and the logout click event from the parent. Now this function and event are in this file))
Picture1

vladhybalo commented on src/ducks/userInfo/userInfoActions.jsx:
```diff
@@ -79,4 +87,17 @@ export const fetchGoogleData = () => {
+export const fetchLogout = () => {
+    return dispatch => {
+        axios
+            .post(`https://kidslike-v1-backend.goit.global/auth/logout`)
```
I think you need to pass token into header, otherwise server wont be able to logout you

DmytroP888 replied:
Yes, you are right)) Here is such an option, there seems to be no error and the request works. But I'm still not sure if this is correct...
![image](https://user-images.githubusercontent.com/96341034/189653895-ffdc33bc-d9d2-47d8-a4df-4ba930e20bc0.png)

VladyslavHybalo replied:
You should pass token as Authorization header, not like body

DmytroP888 replied:
It seems to me that this is how you can specify the request header. However, I still look forward to your review?))
![image](https://user-images.githubusercontent.com/96341034/189667314-16e4743c-c7e3-4490-a699-494026dab246.png)

VladyslavHybalo replied:
where do you get user info here? Its not defined

DmytroP888 replied:
Can I create a constant and use it?

export const fetchLogout = () => {
    const userAuthorized = store.getState().userInfo.token
    return dispatch => {
        axios
            .post(`https://kidslike-v1-backend.goit.global/auth/logout`, {
                headers: {
                    'Authorization': userAuthorized
                }
            })
=================================================
Or can specify a value in - payload?

vladhybalo commented on src/components/Header/HeaderUser.jsx:
```diff
@@ -0,0 +1,99 @@
+
+const HeaderUser = () => {
+    const dispatch = useDispatch()
+    const logout = () => dispatch(fetchLogout(formData))
```
Looks like you didnt define formData var and we get an error into console that could be probably a crash)
![image](https://user-images.githubusercontent.com/52242019/189525835-4c0154f2-6bf7-4d5a-85fe-bac58cef7fdf.png)
PS you didnt need it at all

DmytroP888 replied:
Thank you) I removed this argument
